src/pages/Swap/index.tsx

```diff
@@ -36,12 +36,25 @@ import Loader from 'components/Loader'
 import useI18n from 'hooks/useI18n'
 import PageHeader from 'components/PageHeader'
 import ConnectWalletButton from 'components/ConnectWalletButton'
+import V2ExchangeRedirectModal from 'components/V2ExchangeRedirectModal'
 import AppBody from '../AppBody'
 
+const StyledLink = styled(Link)`
+  display: inline;
+  color: ${({ theme }) => theme.colors.failure};
+`
+
 const Swap = () => {
   const loadedUrlParams = useDefaultsFromURLSearch()
   const TranslateString = useI18n()
-
+  const [modalCountdownSecondsRemaining, setModalCountdownSecondsRemaining] = useState(5)
+  const [disableSwap, setDisableSwap] = useState(false)
+  const [hasPoppedModal, setHasPoppedModal] = useState(false)
+  const [interruptRedirectCountdown, setInterruptRedirectCountdown] = useState(false)
+  const [onPresentV2ExchangeRedirectModal] = useModal(
+    <V2ExchangeRedirectModal handleCloseModal={() => setInterruptRedirectCountdown(true)} />
```

Comment:
Interrupt the countdown when the modal is closed, preventing a redirect and triggering the condition on [ln 129](https://github.com/pancakeswap/pancake-swap-interface-v1/pull/418/files#diff-d2d2d34074b22ac06988a155c74cca892a97f5bb4e343d933a1a4f357d4bf888R129)